### DISCUSSION TITLE
Unable to install Legacy Updater

### DISCUSSION THREAD

**Discussion by DilanRona** [LegacyUpdate/LegacyUpdate#300] [Questions & Help]
Unable to install legacy updater. Error message given is "Component store has been corrupted".

This is not the fault of Legacy updater, but of the Windows 8.1 os I installed (fresh), all attempts to repair it has failed. What do I do when the sources for the dism command is no longer valid?

**Comment by kirb** [comment]
Since you’re using Windows 8(.1), you can try the “Refresh your PC” feature. [Here are Microsoft’s instructions](https://support.microsoft.com/windows/how-to-refresh-reset-or-restore-your-windows-8-1-pc-51391d9a-eb0a-84a7-69e4-c2c1fbceb8dd). This downloads and reinstalls the OS via Windows Update. Note - don’t confuse **refresh** and **reset**. Refresh means to reinstall the OS, while reset means to erase your data.

You can also do a repair reinstall using your Windows DVD. Insert/mount it, and double-click setup.exe. When prompted, you’ll need to make sure one of the following are selected:

* Windows Vista/7: Which type of installation do you want?: “Upgrade”
* Windows 8/8.1: Choose what to keep: “Keep Windows settings, personal files, and apps”
* Windows 10/11: Choose what to keep: “Keep personal files and apps”

(I added the others for anyone else coming across this thread, since they can all use the same process.)

If this option isn’t selectable, you’re using the wrong DVD. It needs to be the same bitness (32/64), and the same or higher version and edition.
